### PR TITLE
Set the PATH env variable for 'type' builtin.

### PR DESCRIPTION
**Pull request by lonicerae**
- See discussion in 5817b14c86e15a2ea72f2a842010458994ef19a7

### CODE REVIEW

**Comment by lonicerae** [comment]
@masterkorp Could you please help to review this PR? Thanks!

**Comment by masterkorp** [comment]
@lonicerae Thanks for the submission, although I do that #21 solves this PATH problem, I am not able to test the situation.

File: update-resolv-conf.sh [hunk 1]
```diff
@@ -17,7 +17,11 @@
 # foreign_option_3='dhcp-option DOMAIN be.bnc.ch'
 # foreign_option_4='dhcp-option DOMAIN-SEARCH bnc.local'
 
-## You might need to set the path manually here, i.e.
+## The 'type' builtins will look for file in $PATH variable, so we set the
+## PATH below. You might need to directly set the path to 'resolvconf'
+## manually if it still doesn't work, i.e.
+## RESOLVCONF=/usr/sbin/resolvconf
+export PATH=$PATH:/sbin:/usr/sbin:/bin:/usr/bin
```
Comment (masterkorp) [inline update-resolv-conf.sh]: Why the use of export here? Would it be?

Comment (lonicerae) [inline update-resolv-conf.sh]: The thought originally came from my local bash configuration. 'export'ing a variable can make it available to other programs, a quick ref:

- http://stackoverflow.com/questions/7411455/what-does-export-do-in-shell-programming

I tried setting the 'PATH' var without exporting it in the 'openvpn-update-resolv-conf' script just now, it also worked without problem, as far as I could see the 'resolvconf' script can work without throwing errors. We can go ahead to remove the 'export' directive.

Comment (masterkorp) [inline update-resolv-conf.sh]: @lonicerae exporting will make the variable available to other programs but only on that bash proccess, and openvpn call a bash for it.

**Comment by masterkorp** [comment]
Can you test the master branch now that #21 was merged?

File: update-resolv-conf.sh (hunk 1 above)
Comment (lonicerae) [inline update-resolv-conf.sh]: Hi @masterkorp , actually that commit won't resolve the problem, please see the log below:
~~~
Fri Mar 17 09:54:37 2017 Data Channel Encrypt: Cipher 'AES-256-CBC' initialized with 256 bit key
Fri Mar 17 09:54:37 2017 Data Channel Encrypt: Using 160 bit message hash 'SHA1' for HMAC authentication
Fri Mar 17 09:54:37 2017 Data Channel Decrypt: Cipher 'AES-256-CBC' initialized with 256 bit key
Fri Mar 17 09:54:37 2017 Data Channel Decrypt: Using 160 bit message hash 'SHA1' for HMAC authentication
Fri Mar 17 09:54:37 2017 Control Channel: TLSv1.2, cipher TLSv1/SSLv3 DHE-RSA-AES256-GCM-SHA384, 2048 bit RSA
Fri Mar 17 09:54:37 2017 [ovpn-srv.example.com] Peer Connection Initiated with [AF_INET]119.254.120.114:443
Fri Mar 17 09:54:39 2017 SENT CONTROL [ovpn-srv.example.com]: 'PUSH_REQUEST' (status=1)
Fri Mar 17 09:54:39 2017 PUSH: Received control message: 'PUSH_REPLY,route-gateway 10.71.6.1,route 10.0.0.0 255.0.0.0,dhcp-option DNS 10.72.6.26,dhcp-option DNS 10.35.255.14,dhcp-option DOMAIN example.com,ping 30,ping-exit 600,explicit-exit-notify 3,topology subnet,ifconfig 10.71.6.64 255.255.252.0'
Fri Mar 17 09:54:39 2017 OPTIONS IMPORT: timers and/or timeouts modified
Fri Mar 17 09:54:39 2017 OPTIONS IMPORT: --explicit-exit-notify can only be used with --proto udp
Fri Mar 17 09:54:39 2017 OPTIONS IMPORT: --ifconfig/up options modified
Fri Mar 17 09:54:39 2017 OPTIONS IMPORT: route options modified
Fri Mar 17 09:54:39 2017 OPTIONS IMPORT: route-related options modified
Fri Mar 17 09:54:39 2017 OPTIONS IMPORT: --ip-win32 and/or --dhcp-option options modified
Fri Mar 17 09:54:39 2017 ROUTE_GATEWAY 192.168.17.1/255.255.255.0 IFACE=enp0s25 HWADDR=28:d2:44:66:ea:0f
Fri Mar 17 09:54:39 2017 TUN/TAP device tun0 opened
Fri Mar 17 09:54:39 2017 TUN/TAP TX queue length set to 100
Fri Mar 17 09:54:39 2017 do_ifconfig, tt->ipv6=0, tt->did_ifconfig_ipv6_setup=0
Fri Mar 17 09:54:39 2017 /usr/sbin/ip link set dev tun0 up mtu 1360
Fri Mar 17 09:54:39 2017 /usr/sbin/ip addr add dev tun0 10.71.6.64/22 broadcast 10.72.11.255
Fri Mar 17 09:54:39 2017 /etc/openvpn/update-resolv-conf.sh tun0 1360 1419 10.71.6.64 255.255.252.0 init
dhcp-option DNS 10.72.6.26
dhcp-option DNS 10.32.2.13
dhcp-option DOMAIN example.com
>>>>> /etc/openvpn/update-resolv-conf.sh: line 60: -x: command not found <<<----------------- ERROR HERE
Fri Mar 17 09:54:39 2017 /usr/sbin/ip route add 10.0.0.0/8 via 10.71.6.1
Fri Mar 17 09:54:39 2017 Initialization Sequence Completed
^CFri Mar 17 09:55:06 2017 event_wait : Interrupted system call (code=4)
Fri Mar 17 09:55:06 2017 /usr/sbin/ip route del 10.0.0.0/8
Fri Mar 17 09:55:06 2017 Closing TUN/TAP interface
Fri Mar 17 09:55:06 2017 /usr/sbin/ip addr del dev tun0 10.71.6.64/22
Fri Mar 17 09:55:06 2017 /etc/openvpn/update-resolv-conf.sh tun0 1360 1419 10.71.6.64 255.255.252.0 init
>>>>> /etc/openvpn/update-resolv-conf.sh: line 63: -d: command not found <<<----------------- ERROR HERE
Fri Mar 17 09:55:06 2017 SIGINT[hard,] received, process exiting
~~~
The root cause I've mentioned in our previous discussion, changing the shell bang can just enable you to find the 'bash' shell, but not changing the searching path for searching the **resolvconf**. That's why I proposed to change the PATH variable. Especially, even if you export the PATH in the update-resolv-conf.sh, it won't pollute the system environment, as the effect will be unavailable once the update-resolv-conf.sh and inner scripts/commands terminate or abort.
I don't mean to push you to adopt my patch, but we should find out what's the real problem before we solve it. Have a nice day.

**Comment by lonicerae** [comment]
I've tested with the latest commit, please see my reply above.

**Pull request merged**